hpx/lcos/local/latch.hpp

```diff
@@ -131,15 +134,18 @@ namespace hpx { namespace lcos { namespace local
         ///
         void wait() const
         {
-            std::unique_lock<mutex_type> l(mtx_);
-            if (counter_ > 0)
-                cond_.wait(l, "hpx::local::latch::wait");
+            if (counter_.load(std::memory_order_acquire) > 0)
+            {
+                std::unique_lock<mutex_type> l(mtx_.data_);
+                if (counter_.load(std::memory_order_relaxed) > 0)
+                    cond_.data_.wait(l, "hpx::local::latch::wait");
```

Comment:
A loop might have been better here:
```
while (counter_.load(std::memory_order_acquire) > 0)
{
    std::unique_lock<mutex_type> l(mtx_.data_);
    cond_.data_.wait(l, "hpx::local::latch::wait");
}

Reply:
Could you elaborate, please? I don't see the use case for this. If the `counter_` reaches zero all threads should be notified once. There does not seem to be a need for the loop.